app/controllers/user.js

```diff
@@ -479,5 +496,47 @@ module.exports = {
     } catch (error) {
       HANDLER.handleError(error)
     }
+  },
+  // Find User and Create if doesn't exist
+  findOrCreateForOAuth: async(profile, provider) => {
+    let user;
+    if(provider==='google'){
+      user = formatUser.formatUser(profile, provider)
+    }
+    try {
+      const existingUser = await User.findOne({
+        email: user.email
+      })
+      if (existingUser) {
+        const token = await existingUser.generateAuthToken()
+        return  {token, user: existingUser};
```

Comment:
why returning, instead of sending a response. Any specific reason?

Reply:
Because we're handling response from OAuth callback route(`/auth/google/callback`).
I don't think it'll be appropriate to send a response from here.